tensorflow/python/kernel_tests/map_ops_test.py

```diff
@@ -161,11 +158,27 @@ def testMultipleInsertLookupGrad(self):
       l2 = map_ops.tensor_map_lookup(m, k2, v2.dtype)
       l3 = map_ops.tensor_map_lookup(m, k3, v3.dtype)
       g = tape.gradient(l * 5, v)
-      self.assertAllClose(g, 5)
       g2 = tape.gradient(l2 * 5, v2)
-      self.assertAllClose(g2, 5)
       g3 = tape.gradient(l3 * 5, v3)
+      self.assertAllClose(g, 5)
+      self.assertAllClose(g2, 5)
       self.assertAllClose(g3, 5)
 
+  def testSameKeyInsertLookupGrad(self):
+    with backprop.GradientTape(persistent=True) as tape:
+      m = map_ops.empty_tensor_map()
+      k = constant_op.constant(1.0)
+      v = constant_op.constant(2.0)
+      v2 = constant_op.constant(22.0)
+      tape.watch(v)
+      tape.watch(v2)
+      m = map_ops.tensor_map_insert(m, k, v)
+      m = map_ops.tensor_map_insert(m, k, v2)
+      l = map_ops.tensor_map_lookup(m, k, v.dtype)
```

Comment:
Can we also do the converse:

insert(m, k, v)
l1 = lookup(m, k, dt)
# check gradient(l1, v)

insert(m, k, v2)
l2 = lookup(m, k, dt)
# check gradient(l2, v2)
# check gradient(l2, v)

Reply:
done!